### ISSUE TITLE
Button does not use the CSS class

### ISSUE THREAD

**Issue by Minerik007** [linkfrg/ignis#52]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

No

### System Information

 ```
Ignis 0.3.dev0
Branch: main
Commit: b9fa6b3858650b3dd53ccee3266db4a49bcaf9c5 (version: bump to dev)
Current desktop: sway

os-release:
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo
```

It says sway but i am on Hyprland

### Description

I cant change style of an button with CSS class. It just doesn't do anything. I tried example of simple bar and it has same issue.

### How to reproduce

Just download and run simple bar example.

### Logs, images, videos, config files, CSS/SCSS files

![image](https://github.com/user-attachments/assets/1cb4fef0-5902-44b4-844d-3b4f5228e9ab)

**Comment by linkfrg** [comment]
Please share your (modified) config file and ``style.scss``

**Comment by Minerik007** [comment]
Problem is that it is not modified. But there it is.
[config.py.txt](https://github.com/user-attachments/files/18014017/config.py.txt)
[style.scss.txt](https://github.com/user-attachments/files/18014018/style.scss.txt)

**Comment by linkfrg** [comment]
Tried on the example bar

style.scss
```scss
.clock {
    font-size: 1.1rem;
    margin-right: 0.5rem;
    background-color: red;
}
```
And this changed the clock color to red:
![image](https://github.com/user-attachments/assets/49d0bc5e-29c5-4c81-9290-f81d2a35562a)

**Comment by Minerik007** [comment]
Yeah with every non-interactible element it works. But with buttons, sliders etc. no

For example look at workspace buttons on my screenshot vs on yours

**Comment by linkfrg** [comment]
> Yeah with every non-interactible element it works. But with buttons, sliders etc. no
> 
> For example look at workspace buttons on my screenshot vs on yours

```scss
.workspace {
    border-radius: 1rem;
    padding: 0 0.55rem;
    transition: 0.3s;
    background-color: red;
}
```

![image](https://github.com/user-attachments/assets/b1fc3b70-d464-453c-9cad-82fdb0b83097)

**Comment by Minerik007** [comment]
but why it doesnt work for me

**Comment by Minerik007** [comment]
> ~/.config/gtk-4.0/gtk.css

Can it be fixed without deleting this file?

**Comment by linkfrg** [comment]
> > ~/.config/gtk-4.0/gtk.css
> 
> Can it be fixed without deleting this file?
> 
> 

Hmmm, I have one idea, it requires small change to Ignis

**Comment by Minerik007** [comment]
> Hmmm, I have one idea, it requires small change to Ignis

Can you do that?
Btw shouldn't `* { all: unset; }` distable all system styles? If yeah, it dont seem to work. Like you can see.

**Comment by Minerik007** [comment]
> Gtk has different priorities for style providers (Gtk.CssProvider): [pgi docs](https://lazka.github.io/pgi-docs/index.html#Gtk-4.0/constants.html#Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION)

Thanks to this i just found out how to fix it.
I needed to add `env = GTK_THEME,Catppuccin-Mocha-Standard-Blue-Dark
` to the Hyprland config and delete `~/.config/gtk-4.0/gtk.css`.
Now it sets my GTK theme correctly to applications and in the same time Ignis works.
I didnt know that `~/.config/gtk-4.0/gtk.css` has higher priority than normal theme.
Thanks for help. I am closing this issue. Have an great day :)

**Comment by jerbaroo** [comment]
I am using home-manager to set my GTK theme. I don't believe there is an option with home-manager to install a GTK theme without setting `~/.config/gtk-4.0/gtk.css`.

Solution that worked for me:
```py
app = IgnisApp.get_default()
app.widgets_style_priority = "user"
```
